Neos.Fusion/Classes/FusionObjects/ComponentImplementation.php

```diff
@@ -55,6 +57,10 @@ public function evaluate()
     protected function prepare(array $context): array
     {
         $context['props'] = $this->getProps($context);
+        $private = $this->getPrivateProps($context);
+        if ($private !== null) {
+            $context['private'] = $private;
```

Comment:
we should consider always setting the context `private` (at leasz to null) so there is a consistent behavior that outer private props will never pass.

or maybe less breaking: check if `private` is instance of lazyProps and disallow only that